geo-traits/src/coord.rs

```diff
@@ -0,0 +1,78 @@
+use geo_types::{Coord, CoordNum, Point};
+
+pub trait CoordTrait: Send + Sync {
+    type T: CoordNum + Send + Sync;
+
+    /// x component of this coord
+    fn x(&self) -> Self::T;
+
+    /// y component of this coord
+    fn y(&self) -> Self::T;
+
+    /// Returns a tuple that contains the x/horizontal & y/vertical component of the coord.
+    fn x_y(&self) -> (Self::T, Self::T);
```

Comment:
Do you think it's reasonable to have a default implementation for this so implementers don't need to implement it?